### PR TITLE
#358 file name change when downloading csv

### LINKED ISSUE #358: gradebook tools download file name needs to be fixed
When the gradebook file is generated, its name is "third_party_gradebook_puff.csv." or "Canvas_gradebook_geff.csv

This name needs to be fixed to not include "puff." It would seem the most useful name for the user would be some combination of the original filename and the words "formatted" or "canavas-formatted." 

Whatever we decide, it should not include the word "puff/geff" which is just an internal acronym for one third party tool.

### PR DESCRIPTION
**Pull request by pushyamig**
Fixes #358 

This affects both Gradebook tools, not just 3rd-party gradebook

### CODE REVIEW

File: ccm_web/client/src/utils/fileUtils.ts [hunk 1]
```diff
@@ -5,14 +5,18 @@ const prepDownloadDataString = (data: string): string => {
...
 Examples:
-  createOutputFileName('old-file-name.csv', '-puff') returns 'old-file-name-puff.csv'
-  createOutputFileName('old-file-name', '-puff') returns 'old-file-name-puff
-  createOutputFileName('old.file.name.csv', '-puff') returns 'old.file.name-puff.csv'
+  createOutputFileName('old-file-name.csv', '-new') returns 'old-file-name-new.csv'
+  createOutputFileName('old-file-name') returns 'old-file-name-formatted
+  createOutputFileName('old.file.name.csv', '-alt') returns 'old.file.name-alt.csv'
 */
-function createOutputFileName (oldFileName: string, newfileEnding: string): string {
+function createOutputFileName (oldFileName: string, newfileEnding?: string): string {
```
Comment (pushyamig) [inline ccm_web/client/src/utils/fileUtils.ts:16]: I made this optional so we can have a choice for a future feature download as needed

File: ccm_web/client/src/utils/fileUtils.ts [hunk 2]
```diff
@@ -5,14 +5,18 @@ const prepDownloadDataString = (data: string): string => {
...
+  if (newfileEnding === undefined) {
+    newfileEnding = '-formatterd'
```
Comment (ssciolla) [inline ccm_web/client/src/utils/fileUtils.ts:18]:
```suggestion
    newfileEnding = '-formatted'
```

File: ccm_web/client/src/utils/fileUtils.ts [hunk 3]
```diff
@@ -5,14 +5,18 @@ const prepDownloadDataString = (data: string): string => {
 }
 
 /*
-Takes a file name and replaces the string segment without a period
+Takes a file name and replaces the string segment(as optional) without a period
 just before the file name extension with that segment plus a new provided ending.
+the default string segment is '-formatted'
```
Comment (ssciolla) [inline ccm_web/client/src/utils/fileUtils.ts:10]:
```suggestion
Takes a file name and replaces the string segment without a period
just before the file name extension with that segment plus a new ending
(which can be optionally provided). The default new ending is '-formatted'.
```

**Review by ssciolla — APPROVED** [review]
Looks fine. I suggested some changes to the comment.

**Review by ssciolla — APPROVED** [review]

**Pull request merged**